### ISSUE TITLE
Converting a trained model for inference but using Training=True

### ISSUE THREAD

**Issue by dennex** [openvinotoolkit/openvino#11445]
I have trained a Pix2Pix network using Keras Tensorflow, following this tutorial. The Pix2Pix uses Instance Normalization, such that when doing inference, we would need to have the Instance Normalization layers (batch norm for batch size of 1) to compute the sample mean and variance. In Tensorflow, I would call the forward as pred = model(x, training=True). The model is the generator part of the Pix2Pix, which is a UNet with Instance Normalization.

```
model = tf.keras.models.load_model("pix2pix")
pred = model(img, training=True)
```

https://www.tensorflow.org/tutorials/generative/pix2pix

We are using this model in C++ using the OpenCV DNN to perform inference, and we are using OpenVINO to convert a TF "saved_model" into Intermediate Representation for being loaded by OpenCV DNN.

The problem is that when doing inference, the Batch normalizations layers use the trained statistics to do normalization, whereas we would want the u and sigma to be calculated "per instance" even in inference. Is there a way to convert a saved_model into intermediate representation keeping the "training=True"?

**Comment by rkazants** [comment]
@dennex, can you provide your trained model please? I will take a look. Sometimes the saved model contains operations with `is_training=True` and MO converts these operations as is, taking this attribute into account.

**Comment by dennex** [comment]
[saved_model.zip](https://github.com/openvinotoolkit/openvino/files/8420015/saved_model.zip)
Here's my TF saved model. You'll need to change the extension to .pb

I've actually found the same thing:
- converted to pbtxt for readability, looking through the file, I see layers of FusedBatchNormV3 with the is_training parameter set to false. 
`attr {
          name: "is_training"
          type: "bool"
          default_value {
            b: false
          }
        }`

- change them all to true
- feed the modified .pbtxt to model optimizer and it compiles a set of saved_model.bin and .xml which actually retains these as true, and we can use these to perform inference with the BatchNorms being active.

Thanks!

**Comment by rkazants** [comment]
@dennex, I think we can close this ticket because you have seemed to find a workaround. Now there is no possibility to specify `is_training` value through MO for TF2 models. 
It looks like more a feature request. Do you know how often this feature is needed?

Best regards,
Roman

**Comment by dennex** [comment]
Agreed, I was hoping it was a hidden feature I was not able to find in the documentation.
How often:
- using a model in inference with "training=True" seems recommended everywhere where instance normalization is used, which is recognized as an improvement for style transfer and image translation GANs. As those types of models get more popular and move from an R&D environment to field applications where we want a lightweight "inference" in C++, this feature will become more important.
This should be both in the model optimizer as an option, and in the inference engine.

We are okay to close this ticket though!

Thanks
